### PR TITLE
COM-486: Prevent XSS attacks in `isLinkTarget` validator

### PR DESCRIPTION
**Pull request by jomunker**
## Description
This PR adds a filter for `javascript` and `data` in the `isLinkTarget` function and decorator to prevent XSS attacks.

### CODE REVIEW

**Review by max-debug022 — APPROVED** [review]

**Comment by johnnyomair** [comment]
I know this is a bit out of scope, but could you please rename the files to the new name? This was forgotten in #1671.

**Comment by johnnyomair** [comment]
Also, please add a changeset.

**Comment by jomunker** [comment]
> Also, please add a changeset.

Sure. What should be the content of that?

**Comment by johnnyomair** [comment]
> > Also, please add a changeset.
> 
> Sure. What should be the content of that?

IMO it should be a patch. Maybe something like:

```md
Prevent XSS attacks in `isLinkTarget` validator
```

**Comment by jomunker** [comment]
@johnnyomair should isLinkTarget be exported in the index file?

File: packages/api/blocks-api/src/blocks/externalLinkBlock/is-href.validator.ts [hunk 1]
```diff
@@ -27,7 +27,11 @@ export class IsLinkTargetConstraint implements ValidatorConstraintInterface {
             return false;
```
Comment (johnnyomair) [inline packages/api/blocks-api/src/blocks/externalLinkBlock/is-href.validator.ts:27]: Please also rename this file and remove the unnecessary deprecations, thanks.

**Review by johnnyomair — APPROVED** [review]

**Review by max-debug022 — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**